### PR TITLE
(DE-735) feat(dbt): add proportion_test and move tracking test to start of the transfo

### PR DESCRIPTION
**Pull request by LucileRainteau**
# DE PR

## Describe your changes

Please include a summary of the changes:

* This PR [adds/removes/fixes/replaces] the [feature/bug/etc].
* Tag a reviewer if necessacy  @github/username

## Jira ticket number and/or notion link

JIRA-ticket_number

### Type of change

* [ ] Fix (non-breaking change which corrects expected behavior)
* [ ] New fields (non-breaking change)
* [ ] New table (non-breaking change)
* [ ] Concept change (potentially breaking change which modifies fields according to new or evolving business concepts)
* [ ] Table deletion (potentially breaking change which adds functionality/ table)

### Checklist before requesting a review

* [ ] I have performed a self-review of my code

* [ ] My code passes CI/CD tests
* [ ] I updated README.md
* [ ] I have updated the dag
* [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
* [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
* [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
* [ ] I will create a review on slack and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### Added tests?

* [ ] 👍 yes
* [ ] 🙅 no, because they aren't needed
* [ ] 🙋 no, because I need help
* [ ] ⏰ no, but I created a ticket

### PR title format (except for MEP)

(ticket) type(topic): comment

with:

* ticket surrounded by parenthesis, with optionnaly a hyphen followed by one or more digits (e.g., -1234). The first part must be one of the following strings:
  * DA
  * DE
  * AE
  * DS
  * HF
  * BSR
  * PC

* type :
The second part to specify the type of change one of the following :
  * build
  * lint
  * ci
  * docs
  * feat
  * fix
  * perf
  * refactor
  * test
  * core
  * dbt

* topic within parenthesis: 1 word e.g., (dag)

* comment: tell us your life

examples:

* :white_check_mark: (DE-124) refactor(firebase): update source field
* :x: (DE-124) refactor (firebase): update source field **(space between type and topic)**
* :x: (DE-124) airflow(firebase): update source fiedd in DAG **(wrong type)**
* :x: (DE-124) (DE-124) refactor(firebase refacto): update source field **(topic in two words)**
* :white_check_mark: (BSR) docs(github): add PR title valid format in template

### CODE REVIEW

File: orchestration/dags/data_gcp_dbt/macros/custom_test/not_null_proportion_test.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        ),
+        final as (
+            select
+                cast(matching_records as float64)
+                / nullif(total_records, 0) as proportion
+            from validation
+        )
+    select *
+    from final
+    where proportion > 0.01
```
Comment (valoumiaou) [inline orchestration/dags/data_gcp_dbt/macros/custom_test/not_null_proportion_test.sql:21]: variable dans le dbt_project ?

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event_flattened.sql [hunk 2]
```diff
@@ -146,5 +146,14 @@ select
         venues_41_50
     ),
     {{ extract_str_to_array_field("offers", 0, 10, 50) }} as displayed_offers,
-    {{ extract_str_to_array_field("venues", 0, 10, 50) }} as displayed_venues
+    {{ extract_str_to_array_field("venues", 0, 10, 50) }} as displayed_venues,
+    case
```
Comment (valoumiaou) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event_flattened.sql:150]: (event_name = 'BookingConfirmation' and bookingid is null) or (when event_name = 'BookingConfirmation' and user_id is null) as is_anomaly

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event.sql [hunk 3]
```diff
@@ -174,4 +174,4 @@ from {{ ref("int_firebase__native_event_flattened") }} as e
     where
         event_date
         between date_sub(date("{{ ds() }}"), interval 3 day) and date("{{ ds() }}")
-{% endif %}
+{% endif %} and is_anomaly = 0
```
Comment (valoumiaou) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event.sql]: and not is_anomally

File: orchestration/dags/data_gcp_dbt/macros/custom_test/not_null_proportion_test.sql [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{% test not_null_proportion(model, column_name, where_condition, warn_if) %}
+
+    with
+        validation as (
+            select
+                count(*) as total_records,
+                sum(
```
Comment (valoumiaou) [inline orchestration/dags/data_gcp_dbt/macros/custom_test/not_null_proportion_test.sql:7]: count(CASE WHEN NOT {{ column_name }} THEN 1 END) as matching_records

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/schema/int_firebase__native_event_flattened.yaml [hunk 5]
```diff
@@ -19,15 +20,19 @@ models:
         data_tests:
           - not_null:
               where: "event_date >= DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY)"
-      - name: booking_id
+      - name: user_id
+        data_tests:
+          - not_null_proportion:
+              where_condition: "event_date >= DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY) and event_name = 'BookingConfirmation'"
```
Comment (valoumiaou) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/schema/int_firebase__native_event_flattened.yaml:26]: je ne sais plus si on avait dit 1 jour ou plus

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/schema/int_firebase__native_event_flattened.yaml [hunk 6]
```diff
@@ -19,15 +20,19 @@ models:
...
         data_tests:
           - relationships:
               to: ref("int_global__booking")
               field: booking_id
               where: "event_date >= DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY)"
-          - not_null:
-              where: "event_name = 'BookingConfirmation' AND event_date >= DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY)"
-      - name: offer_id
+          - not_null_proportion:
+              where_condition: "event_name = 'BookingConfirmation' AND event_date >= DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY)"
```
Comment (valoumiaou) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/schema/int_firebase__native_event_flattened.yaml:34]: idem

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/schema/int_firebase__native_event_flattened.yaml [hunk 7]
```diff
@@ -19,15 +20,19 @@ models:
...
+      - name: offerId
         data_tests:
-          - not_null:
-              where: "event_name = 'ConsultOffer' AND event_date >= DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY)"
+          - not_null_proportion:
```
Comment (valoumiaou) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/schema/int_firebase__native_event_flattened.yaml:37]: idem

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/schema/int_firebase__native_event_flattened.yaml (hunk 5 above)
Comment (LucileRainteau) [reply orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/schema/int_firebase__native_event_flattened.yaml:26]: On fait les tests sur la veille mais ils sont stockés dans la table elementary_test_results
Elementary montre l'historique d'execution des tests sur les 7 derniers jours

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event_flattened.sql (hunk 2 above)
Comment (LucileRainteau) [reply orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event_flattened.sql:150]: pourquoi ? pas sure de comprendre
je trouve ça plus facile avec case when pour la lisibilité + pour l'extension des conditions

File: orchestration/dags/data_gcp_dbt/macros/custom_test/not_null_proportion_test.sql (hunk 4 above)
Comment (LucileRainteau) [reply orchestration/dags/data_gcp_dbt/macros/custom_test/not_null_proportion_test.sql:7]: ça marche pas pcq les colonnes testées ne sont pas des bool

File: orchestration/dags/data_gcp_dbt/macros/custom_test/not_null_proportion_test.sql [hunk 8]
```diff
@@ -18,6 +19,6 @@
         )
     select *
     from final
-    where proportion > 0.01
+    where proportion > {{ threshold_test }}
```
Comment (valoumiaou) [inline orchestration/dags/data_gcp_dbt/macros/custom_test/not_null_proportion_test.sql:22]: pourquoi on set threshold_test avant en variable?
ca marche pas ?
{{ var("threshold_test") }}

File: orchestration/dags/data_gcp_dbt/dbt_project.yml [hunk 9]
```diff
@@ -117,3 +117,4 @@ vars:
   region_name: region-europe-west1
   mute_ensure_materialization_override: true
   snapshot_filter: "dbt_valid_to is null"
+  threshold_test: 0.01
```
Comment (valoumiaou) [inline orchestration/dags/data_gcp_dbt/dbt_project.yml:120]: maybe un nom plus explicite genre volume_anomaly_alert_threshold
?

**Review by valoumiaou — APPROVED** [review]

**Pull request merged**